src/test/java/vinegar/VinegarTest.java

```diff
@@ -0,0 +1,99 @@
+package vinegar;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import vinegar.storage.Storage;
+import vinegar.task.TaskList;
+import vinegar.ui.Ui;
+import vinegar.command.Command;
+import vinegar.task.Todo;
+import vinegar.task.Deadline;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class VinegarTest {
+
+    private Vinegar vinegar;
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+
+    @TempDir
+    Path tempDir; // Temporary directory for testing
+
+    @BeforeEach
+    public void setUp() {
+        String filePath = tempDir.resolve("vinegar.txt").toString();
+        vinegar = new Vinegar(filePath);
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList();
+    }
+
+    @Test
+    public void testRunCommand_AddTodo() throws IOException, VinegarException {
+        // Simulate adding a Todo task
+        Command command = Parser.parse("todo read book");
+        command.execute(tasks, ui, storage);
+
+        // Verify that the task list size increased
+        assertEquals(1, tasks.size());
+
+        // Verify that the task added is of type Todo and the correct description is returned
+        assertTrue(tasks.get(0) instanceof Todo);
+        assertTrue(tasks.get(0).toString().contains("read book"));
+    }
+
+    @Test
+    public void testRunCommand_AddDeadline() throws IOException, VinegarException {
```

Comment:
I like the accurate use of unitBeingTested_descriptionOfTestInputs_expectedOutcome convention!